src/main/webapp/app/shared/table/CdxBiomarkerAssociationTable.tsx

```diff
@@ -0,0 +1,188 @@
+      const getIndications = async () => {
+        const result = await props.getCdxIndications(props.companionDiagnosticDeviceId);
+        setIndications(result);
+      };
+      getIndications();
+    }
+  }, [props.companionDiagnosticDeviceId]);
+
+  const handleDeleteIndication = () => {
+    if (indicationToDeleteId) {
+      props.deleteEntity(indicationToDeleteId);
+    }
+  };
+
+  const handleCancel = () => {
+    setShowModal(false);
+    setIndicationToDeleteId(null);
+  };
+
+  const columns: Column<IDeviceUsageIndication>[] = [
+    {
+      id: 'gene',
+      Header: 'Gene',
+      Cell({
+        cell: {
+          row: { original },
+        },
+      }: {
+        cell: { row: { original: IDeviceUsageIndication } };
+      }): any {
+        return <div>{original.gene?.hugoSymbol}</div>;
+      },
+    },
+    {
+      id: 'alterations',
+      Header: 'Alterations',
+      Cell({
+        cell: {
+          row: { original },
+        },
+      }: {
+        cell: { row: { original: IDeviceUsageIndication } };
+      }): any {
+        return (
+          <>
+            {original.alterations &&
+              original.alterations
+                .map(alteration => alteration.name)
+                .sort()
+                .join(', ')}
+          </>
+        );
+      },
+    },
+    {
+      id: 'cancerType',
+      Header: 'Cancer Type',
+      Cell({
+        cell: {
+          row: { original },
+        },
+      }: {
+        cell: { row: { original: IDeviceUsageIndication } };
+      }): any {
+        return <div>{getCancerTypeName(original.cancerType)}</div>;
+      },
+    },
+    {
+      id: 'drugs',
+      Header: 'Drug',
+      Cell({
+        cell: {
+          row: { original },
+        },
+      }: {
+        cell: { row: { original: IDeviceUsageIndication } };
+      }): any {
+        return (
+          <>
+            {original.drugs &&
+              original.drugs
+                .map(drug => drug.name)
+                .sort()
+                .join(' + ')}
+          </>
+        );
+      },
+    },
+    {
+      id: 'fdaSubmissions',
+      Header: 'FDA Submissions',
+      Cell({
+        cell: {
+          row: { original },
+        },
+      }: {
+        cell: { row: { original: IDeviceUsageIndication } };
+      }): any {
+        return (
+          <>
+            {original.fdaSubmissions &&
+              original.fdaSubmissions
+                .map(sub => getFdaSubmissionNumber(sub.number, sub.supplementNumber))
+                .sort()
+                .join(', ')}
+          </>
+        );
+      },
+    },
+  ];
+
+  if (props.editable) {
+    columns.push({
+      id: 'remove',
+      Header: 'Remove',
+      Cell({
+        cell: {
+          row: { original },
+        },
+      }: {
+        cell: { row: { original: IDeviceUsageIndication } };
+      }): any {
+        return (
+          <>
+            <CompactButton
+              text="Delete"
```

Comment:
Do we need to add the tooltip? I think the trash icon is very self-explanatory .